keps/sig-auth/4940-psa-block-host-field-in-probes/README.md

```diff
@@ -0,0 +1,354 @@
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+We have a `Host` field that can be set from `TCPSocketAction` and
+`HTTPGetAction` fields which are part of the `ProbeHandler` and
+`LifecycleHandler` structs in Kubernetes. The `Host` field is used
+for allowing users to specify another entity other than the
+podIP (which is the default value) to which Kubelet should
+perform probes to. However this opens it up for security attacks
+since the `Host` field can be set to pretty much any value in the
+system including security sensitive external hosts or localhost on
+the node. Kubelet will be probing this set `Host` value which can
+lead to blind SSRF attacks.
+
+### Goals
+
+* Add  Pod Security Admission (PSA) to enable admins to restrict
+  users from creating probes with the `Host` field set.
+* Deprecate `.Host` field from the API by updating the types.go
+  doc string and any documentation updates on the website.
+
+### Non-Goals
+
+N/A
+
+## Proposal
+
+There is a long term plan to deprecate the existing TCP and HTTP probe
+types in the API to replace them with ones with slightly different semantics.
+See [KEP-4559](https://github.com/kubernetes/enhancements/pull/4558) for more
+details. Given the unsolvable security problems with the Host field,
+we do not plan to offer it in the new types.
+
+Meanwhile, the older API is never going to go away. So we also want to
+add PSA to allow admins to be able to restrict users from creating
+probes with the Host field set when using the (about to be deprecated) API.
+This is implemented by [kubernetes PR 125271](https://github.com/kubernetes/kubernetes/pull/125271)
+that does exactly that. Once we add the PSA, we can also deprecate the API
+as part of this KEP itself.
+
+### Risks and Mitigations
+
+There might be users who depend on the `Host` field in
+their existing probes which will continue to work and if
+newly created probes also need the `Host` field to point
+to an external destination then the admin can avoid enforcing
+the PSA to block it.
+
+## Design Details
+
+Add a Baseline APILevel Pod Security Admission policy to allow admins of the
+cluster to block users from setting `.host` field in:
+
+* container.LivenessProbe.ProbeHandler.HTTPGet.Host
+* container.ReadinessProbe.ProbeHandler.HTTPGet.Host
+* container.StartupProbe.ProbeHandler.HTTPGet.Host
+* container.LivenessProbe.ProbeHandler.TCPSocket.Host
+* container.ReadinessProbe.ProbeHandler.TCPSocket.Host
+* container.StartupProbe.ProbeHandler.TCPSocket.Host
+* container.Lifecycle.PostStart.TCPSocket.Host // Deprecated. TCPSocket is NOT supported as a LifecycleHandler and kept for backward compatibility.
+* container.Lifecycle.PreStop.TCPSocket.Host // Deprecated. TCPSocket is NOT supported as a LifecycleHandler and kept for backward compatibility.
+* container.Lifecycle.PostStart.HTTPGet.Host
+* container.Lifecycle.PreStop.HTTPGet.Host
```

Comment:
`container` is not a full field path... explicitly specify initContainer, container, ephemeralContainers

see other examples at https://kubernetes.io/docs/concepts/security/pod-security-standards/#baseline

```
spec.containers[*]...
spec.initContainers[*]...
spec.ephemeralContainers[*]...
...
```

Reply:
I've updated this except the ephemeralcontainers part cause i think probes with ephemeral are not supported according to the types.go file